src/data/submissions.ts

```diff
@@ -0,0 +1,93 @@
+import {createState} from 'state-pool';
+
+import {post} from '@/api';
+
+import type {Form} from './forms';
+
+/**
+ * @see `#/components/schemas/Submission` in the API spec.
+ */
+export interface SubmissionCreateBody {
+  form: string;
+  formUrl: string;
+  anonymous?: boolean;
+  initialDataReference?: string;
+}
+
+/**
+ * @see `#/components/schemas/NestedSubmissionStep` in the API spec.
+ */
+export interface NestedSubmissionStep {
+  id: string;
+  name: string;
+  url: string;
+  formStep: string;
+  isApplicable: boolean;
+  completed: boolean;
+  canSubmit: boolean;
+}
+
+/**
+ * The shape of a submission as returned from a detail/create endpoint.
+ *
+ * @see `#/components/schemas/Submission` in the API spec.
+ */
+export interface Submission {
+  id: string;
+  url: string;
+  form: string;
+  steps: NestedSubmissionStep[];
+  submissionAllowed: 'yes' | 'no_with_overview' | 'no_without_overview';
+  isAuthenticated: boolean;
+  payment: {
+    isRequired: boolean;
+    amount: null | string; // decimal serialized to string
+    hasPaid: boolean;
+  };
+  formUrl: string;
+  initialDataReference: string;
+}
+
+/**
+ * Create a submission instance from a given form instance
+ * @param baseUrl - The Open Forms backend baseUrl of the API (e.g. https://example.com/api/v2/)
+ * @param form - The relevant Open Forms form instance.
+ * @param formUrl - The client-side URL hosting the form entrypoint.
+ * @param initialDataReference - The data reference provided by the external party.
+ * @return - The Submission instance.
+ */
+export const createSubmission = async (
+  baseUrl: string,
+  form: Form,
+  formUrl: string,
+  signal: AbortSignal,
+  initialDataReference: string,
+  anonymous = false
+): Promise<Submission> => {
+  const createData: SubmissionCreateBody = {
+    form: form.url,
+    formUrl,
+    anonymous,
+    initialDataReference: initialDataReference ? initialDataReference : undefined,
```

Comment:
Is it important to differentiate between empty string and `undefined`? Otherwise we could just do:
```suggestion
    initialDataReference,
```

Reply:
Because, when fetch the submission, it is returned with a empty string (if undefined) right? (Looking at the `test_retrieve_submission_nothing_submitted` backend test)

Reply:
I'm not sure and I opted for the defensive approach - the current version in `main` does not send the parameter at all to the backend if no reference is provided, and that's replicated with setting it to `undefined`. I don't know what will happen if we send an empty string.

Reply:
ah okay, yeah being save/explicit can't hurt. Okay, lets keep it like this